src/main/java/com/aws/greengrass/tes/TokenExchangeService.java

```diff
@@ -85,14 +85,14 @@ public void postInject() {
 
     @Override
     @SuppressWarnings("PMD.CloseResource")
-    protected void startup() {
+    protected void startup() throws InterruptedException {
         logger.atInfo().addKeyValue(PORT_TOPIC, port).addKeyValue(IOT_ROLE_ALIAS_TOPIC, iotRoleAlias)
                 .log("Attempting to start server at configured port {}", port);
         try {
             validateConfig();
-            server = new HttpServerImpl(port, credentialRequestHandler, executor);
-            server.start();
-            logger.atInfo().log("Started server at port {}", server.getServerPort());
+            server.start(port, credentialRequestHandler);
```

Comment:
Why did you change the signature of `start()`? Keep it just `start()`. Everything which was set in the constructor is fine, leave that alone. Just don't create and start the server in the constructor.